### PR TITLE
OpenID For VP

### PR DESCRIPTION
**Pull request by mepeltier**
This PR adds OID4VP support into the OID4VCI plugin, as well as an overhauled demo that includes that code. We have a few things we plan to take care of prior to taking this out of draft:

- [x] Renaming 
    - Now that this plugin supports more than issuance, we'll be dropping the `i` from the plugin name, as well as the current `oid4vci/oid4vci` directory 
- [x] Rebasing/merging in `main`
    - May not even be worth mentioning, but we will make sure to pull in the changes that have happened since we started this work and make sure the PR is able to be merged

### Limitations
- This only supports jwt_vp_json at the moment. An effort similar to #772 is needed to modularize this OID4VP implementation but we think this is a good starting point.
- We plan to update the demo further to more clearly outline the expectations for the controller, providing examples of requests and responses to the Admin API and the pre-conditions for each request. We'll do this in a follow up PR.

### Things that could/should go to ACA-Py

This PR also reimplements some functionality from ACA-Py or adds components that probably belong in ACA-Py:

- JWT signing and verification. The implementation in ACA-Py today is hardcoded to support only EdDSA signatures.
- did:jwk creation
- P-256 key creation and usage
- This one is questionable: Presentation Exchange object evaluation. ACA-Py has an implementation of presentation exchange handling but when we evaluated it, we found it to be not quite what we wanted.

These components can live here until they're added/fixed in ACA-Py. When they are added/fixed in ACA-Py, there shouldn't be a conflict here and we can update the plugin whenever.

### CredProcessor updates

This PR changes the way the modular credential processors are loaded. I was having issues with the old structure because of the file path based dependencies. To avoid that, the `jwt_vc_json` and `mso_mdoc` "lite plugins" have been moved into the oid4vc plugin as additional packages.

The registration and retrieval of these processors has also been updated. The pattern now follows existing patterns in the ACA-Py ecosystem, using a registry-like `CredProcessors` object that is bound to the context. Processors provided by external plugins should register themselves with `CredProcessors` in their setup function. See `mso_mdoc.__init__`'s setup for an example.

The `jwt_vc_json` is active by default. The `mso_mdoc` must be loaded as a plugin (e.g. `--plugin mso_mdoc`) even though it's included in the oid4vc plugin. The `mso_mdoc` extra must also be installed to pull in the `mso_mdoc` dependencies.

### CODE REVIEW

**Comment by dbluhm** [comment]
This PR also reimplements some functionality from ACA-Py or adds components that probably belong in ACA-Py:

- JWT signing and verification. The implementation in ACA-Py today is hardcoded to support only EdDSA signatures.
- did:jwk creation
- This one is questionable: Presentation Exchange object evaluation. ACA-Py has an implementation of presentation exchange handling but when we evaluated it, we found it to be not quite what we wanted.

These components can live here until they're added/fixed in ACA-Py. When they are added/fixed in ACA-Py, there shouldn't be a conflict here and we can update the plugin whenever.

**Comment by dbluhm** [comment]
Heads up, this PR changes the way the modular credential processors are loaded. I was having issues with the old structure because of the file path based dependencies. To avoid that, the jwt_vc_json and mso_mdoc "lite plugins" have been moved into the oid4vc plugin as additional packages.

The registration and retrieval of these processors has also been updated. The pattern now follows existing patterns in the ACA-Py ecosystem, using a registry-like `CredProcessors` object that is bound to the context. Processors provided by external plugins should register themselves with `CredProcessors` in their `setup` function. See `mso_mdoc.__init__`'s setup for an example.

The jwt_vc_json is active by default. The mso_mdoc must be loaded as a plugin (e.g. `--plugin mso_mdoc`) even though it's included in the oid4vc plugin. The mso_mdoc extra must also be installed to pull in the mso_mdoc dependencies.

**Comment by dbluhm** [comment]
cc @weiiv @timbl-ont the above changes may impact your work but I hope you will find this to be an improvement on your modular credential processor contributions!

**Review by mepeltier — COMMENTED** [review]
Mostly minor feedback from me, some cleanup and stuff

**Review by dbluhm — COMMENTED** [review]
We've got a spot of minor clean up to do but I am generally happy with these changes.

**Review by dbluhm — APPROVED** [review]

**Comment by dbluhm** [comment]
I approved but I was involved in this PR so another review from outside of Indicio would be appreciated.

File: oid4vc/README.md [hunk 1]
```diff
@@ -19,41 +19,48 @@ This repository showcases a simplified demonstration of the OID4VCI (OpenID for
...
 ### Demo Flow
 
-Navigate to `http://localhost:3002` in your browser. You will start at the registration page.
+Navigate to `http://localhost:3002` in your browser. You will start at the landing page. The sidebar has buttons to take you to the issuance and presentation pages. 
+
+1. Issue Credential
 
-1. Admin Registration Page
+   - This page generates a simple `UniversityCredential` for issuance
+       - The demo obscures and automate the necessary `credential-supported/create` call, which is what defines the type and values of a credential that can be issued
```
Comment (jamshale) [inline oid4vc/README.md:42]: obscures and ***automates***

File: oid4vc/README.md [hunk 2]
```diff
@@ -150,7 +189,7 @@ The Plugin expects the following configuration options. These options can either
 - `OID4VCI_ENDPOINT` or `oid4vci.endpoint`
     - `credential_issuer` endpoint, seen in the Credential Offer
 - `OID4VCI_CRED_HANDLER` or `oid4vci.cred_handler`
-    - Dict of credential handlers. e.g. `{"jwt_vc_json": "jwt_vc_json.v1_0"}`
+    - Dict of credential handlers. e.g. `{"jwt_vc_json": "jwt_vc_json"}`
```
Comment (jamshale) [inline oid4vc/README.md:193]: LINE 366:
cd oid4vci/int --> cd oidvc/integration

**Review by jamshale — APPROVED** [review]
Wow. A lot of work! I mostly just scanned over it but it looked really good and the documentation was well done.

**Comment by jamshale** [comment]
I'm going to merge this. Looks great!

**Pull request merged**